workflow/util/util.go

```diff
@@ -872,6 +888,17 @@ func retryWorkflow(ctx context.Context, kubeClient kubernetes.Interface, hydrato
 		newWF.Status.StoredTemplates[id] = tmpl
 	}
 
+	if retryArchive {
```

Comment:
behaviour flags like this can be a code smell

Reply:
To address this, we can remove the flag and call `wfClient.create()` in `RetryArchiveWorkflow` after refactoring `retryWorkflow` into `prepareWorkflowForRetry`. This way, we do not need to rely on a flag to determine whether we are retrying an archived workflow or not.